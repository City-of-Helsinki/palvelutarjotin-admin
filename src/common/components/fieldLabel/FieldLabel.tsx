import { Tooltip } from 'hds-react';
import React from 'react';

import styles from './fieldLabel.module.scss';
import { RequiredIndicator } from '../requiredIndicator/RequiredIndicator';

type FieldLabelProps = {
  hidden?: boolean;
  inputId: string;
  label: string | React.ReactNode;
  required?: boolean;
  tooltipLabel?: string;
  tooltipButtonLabel?: string;
  tooltipText?: string;
};

const FieldLabel: React.FC<FieldLabelProps> = ({
  hidden,
  inputId,
  label,
  required,
  tooltipLabel,
  tooltipButtonLabel,
  tooltipText,
  ...rest
}) => (
  <>
    <label
      htmlFor={inputId}
      className={`${styles.label} ${hidden ? styles.hidden : ''}`}
      {...rest}
    >
      {label}
      {required && <RequiredIndicator />}
    </label>
    {tooltipText && (
      <Tooltip
        buttonClassName={styles.tooltipButton}
        tooltipLabel={tooltipLabel}
        buttonLabel={tooltipButtonLabel}
      >
        {tooltipText}
      </Tooltip>
    )}
  </>
);

export default FieldLabel;
