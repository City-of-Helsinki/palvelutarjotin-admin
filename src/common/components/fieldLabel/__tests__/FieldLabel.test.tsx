import { render } from '@testing-library/react';
import React from 'react';

import FieldLabel from '../FieldLabel';

it('matched snapshot', () => {
  const { asFragment } = render(
    <FieldLabel
      hidden={true}
      inputId="test"
      label="foo"
      required
      tooltipLabel="Tooltip label"
      tooltipText="Tooltip text"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
