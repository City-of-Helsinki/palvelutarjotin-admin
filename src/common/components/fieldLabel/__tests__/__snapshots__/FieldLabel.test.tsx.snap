// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matched snapshot 1`] = `
<DocumentFragment>
  <label
    class="label hidden"
    for="test"
  >
    foo
    <span
      aria-hidden="true"
      class="indicator"
    >
      *
    </span>
  </label>
  <div
    class="Tooltip-module_root__1Yo9t"
  >
    <button
      aria-expanded="false"
      aria-label="Tooltip"
      class="Tooltip-module_button__2x9m0 tooltipButton"
      title="Tooltip"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 12v2h-2v-2h2zm-.956-10.25c1.99 0 3.673 1.545 3.673 3.674 0 1.357-.47 2.065-1.556 2.882l-.242.178c-.744.545-.963.852-.98 1.658l-.002.108h-2c0-1.575.522-2.412 1.677-3.288l.248-.182c.706-.517.855-.726.855-1.356 0-.985-.75-1.674-1.673-1.674-.853 0-1.565.59-1.662 1.458l-.009.098-1.996-.114c.115-2.019 1.757-3.442 3.667-3.442z"
              fill="currentColor"
            />
          </g>
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
